### PR TITLE
Added ND-array support

### PR DESCRIPTION
**Pull request by abhishekpratapa**
- Added ND-Array type

### CODE REVIEW

**Comment by abhishekpratapa** [comment]
fixed #442

File: src/unity/lib/unity_sframe.cpp [hunk 1]
```diff
@@ -2004,35 +2013,9 @@ void unity_sframe::explore(const std::string& path_to_client, const std::string&
             }
             break;
           case flex_type_enum::LIST:
-            {
```
Comment (znation) [inline src/unity/lib/unity_sframe.cpp]: What's the deal with all this code being removed? It wasn't needed? Or was it causing bug(s)?

(If the former -- maybe we shouldn't change it now?)
(If the latter -- what bugs was it causing? Should it go in as a separate PR?)

Comment (abhishekpratapa) [reply src/unity/lib/unity_sframe.cpp]: Duplicated code!
